include/silo/common/date.h

```diff
@@ -8,6 +8,8 @@ namespace silo::common {
 
 typedef uint32_t Date;
 
+const Date NULL_DATE = 0;
```

Comment:
Let's also use this in `silo::common::stringToDate`? There are a couple of early returns where this might be appropriate.